docs/rules/no-extra-parens.md

```diff
@@ -13,7 +13,15 @@ A few cases of redundant parentheses are always allowed:
 
 ### Options
 
-The default behavior of the rule is specified by `"all"` and it will report unnecessary parentheses around any expression. The following patterns are considered problems:
+This rule takes 1 argument, a string which must be one of the following:
+
+* `"all"` (default): it will report unnecessary parentheses around any expression.
+* `"all-except-cond-assign"`: extra parens are allowed for assignment in conditional statements.
```

Comment:
We generally don't have long string options like as they get a bit hairy. Any more than one dash is pretty hard on the eyes. We should probably add this as an options object so you can do:

```
no-extra-parens: [2, "all", {
    conditionalAssign: false
}];
```

Or something like that.